radareorg/radare2
# Added break to gdbserver vCont and refactored the code to use locks

The previous gdbr implementation didn't allow interrupting background tasks with &b since send_vcon wasn't properly configured with the cons api. In addition to that, gdbserver doesn't support processing multiple commands at the same time, resulting in undefined behavior once cons were set up for vcont.
This PR adds the relevant cons api and solves the concurrency issues by adding locks on all socket related logic.

I went ahead and fixed style issues and slightly refactored parts of the code. There are still parts that require refactoring from the original implementation but I am not sure if some of the behavior/lack of error checks is intentional. Decided to keep those parts as they were and do a proper refactor once I finish writing tests for gdbserver (incl concurrent tasks).

Testing:

- Execute `dc`, see that you can interrupt it with a regular Ctrl+C
- Execute `& dc`
    - Attempt to run a different command while dc is running with `& [cmd]`, see that you can stop it with `&b [task number]`. It should hang until `dc` is finished unless you interrupt it.
    - Run a command with `& [cmd]` and stop `& dc`'s task number, check that both are done
- Go over debug commands

closes #15415
